src/libraries/System.Private.CoreLib/src/System/Guid.cs

```diff
@@ -801,17 +803,38 @@ public override int GetHashCode()
 
         public bool Equals(Guid g) => EqualsCore(this, g);
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static bool EqualsCore(in Guid left, in Guid right)
         {
-            ref int rA = ref Unsafe.AsRef(in left._a);
-            ref int rB = ref Unsafe.AsRef(in right._a);
+            if (Sse2.IsSupported)
+            {
+                var g1 = Unsafe.As<Guid, Vector128<byte>>(ref Unsafe.AsRef(left));
+                var g2 = Unsafe.As<Guid, Vector128<byte>>(ref Unsafe.AsRef(right));
+
+                if (Sse41.IsSupported)
+                {
+                    var xor = Sse2.Xor(g1, g2);
+                    return Sse41.TestZ(xor, xor);
+                }
+
+                var result = Sse2.CompareEqual(g1, g2);
+                return Sse2.MoveMask(result) == 0b1111_1111_1111_1111;
+            }
+
+            return SoftwareFallback(left, right);
 
-            // Compare each element
+            static bool SoftwareFallback(in Guid left, in Guid right)
+            {
+                ref int rA = ref Unsafe.AsRef(in left._a);
+                ref int rB = ref Unsafe.AsRef(in right._a);
 
-            return rA == rB
-                && Unsafe.Add(ref rA, 1) == Unsafe.Add(ref rB, 1)
-                && Unsafe.Add(ref rA, 2) == Unsafe.Add(ref rB, 2)
-                && Unsafe.Add(ref rA, 3) == Unsafe.Add(ref rB, 3);
+                // Compare each element
+
+                return rA == rB
```

Comment:
Should we special case 64-bits processes by comparing with `long`s?
E.g. on ARM 64 this code will be hit.
Cf. https://github.com/dotnet/runtime/pull/35654

```c#
if (Environment.Is64BitProcess)
{
    // code with long
}
else
{
    // code with int
}
```

The "fast out behavior" should be given with longs too, but I'm not sure how about alignment on ARM 64.

Reply:
I'm happy to update with a special case for 64-bit software fallback if desired. The current `Guid.EqualsCore` method doesn't cater for this, but that's no reason to not cater for it now.